app/assets/javascripts/discourse/controllers/topic-bulk-actions.js.es6

```diff
@@ -26,6 +26,19 @@ export default Ember.Controller.extend(ModalFunctionality, {
   categoryId: Ember.computed.alias('model.category.id'),
 
   onShow() {
+    let showRelistButton = false;
+    const topics = this.get('model.topics');
+    for (let t = 0; t < topics.length; t++) {
+      if (!topics[t].visible) {
+        showRelistButton = true;
+        break;
+      }
+    }
+    if (showRelistButton && !_buttons[9]) {
+      addBulkButton('relistTopics', 'relist_topics');
+    } else if (!showRelistButton && _buttons[9]) {
```

Comment:
Hard coding an index here seems dangerous. What is the condition actually checking? Is it if the button already exists? In that case it should be checking by some kind of indentifier.